roles/oem/tasks/ubuntu_only.yml

```diff
@@ -6,6 +6,13 @@
     upgrade: safe
 - name: Upgrade all installed snaps  # noqa no-changed-when
   command: snap refresh
+- name: Purge old snaps
+  shell: |
+    set -o pipefail && snap list --all | awk '/disabled/{print $1, $3}' |
+    while read snapname revision; do
+      snap remove "$snapname" --revision="$revision"
+    done
```

Comment:
I believe that this will only end up reporting as a failure if the last command in the `while` loop exits non-zero. For example, see:

```bash
set -o pipefail && cat << EOF | while read foo bar; do ( echo $foo $bar; exit $bar ); done
a 1
b 0
c 0
d 0
EOF
```

and the only scenario where this will set `$?` to `0` is if the last line has a non-zero value for the second item. In addition to setting `pipefail` it may be possible to also set `errexit`/`-e`; however, that would result in subsequent revisions not being deleted... The structure also results in additional sub shells due to the pipes.

I'm not 100% sure but perhaps something like:
```suggestion
    errors=0
    while read snapname revision; do
      snap remove "$snapname" --revision="$revision"
      errors=$((errors + $?))
    done <<< "$(snap list --all | awk '/disabled/{print $1, $3}')"
    exit $errors
```

Adding the error counter to the existing while loop (without also switching to a heredoc) resulted in failing to actually count the errors due to the sub shell. I used the following to validate the new suggestion since idk how to actually simulate a `snap remove` failure:

```bash
function create_values() {
cat << EOF
a 1
b 0
c 1
d 0
EOF
}

errors=0
set -o pipefail && while read foo bar; do
    ( echo $foo $bar; exit $bar )
    errors=$((errors + $?))
done <<< "$(create_values)"
exit $errors
```

I also didn't test this on a Ubuntu-ish directly so feel free to tell me if my examples are invalid in different versions of bash (or also if I am wrong generally).

Reply:
Ansible lint still yells if you don't set pipefail, so why not. Also realized while we're exploding the loop, we can just track changes and report based on that instead of forcing it off or on.